src/power_grid_model_io/converters/pandapower_converter.py

```diff
@@ -364,32 +364,40 @@ def _create_pgm_input_lines(self):
             return
 
         switch_states = self.get_switch_states("line")
-        in_service = self._get_pp_attr("line", "in_service", True)
-        length_km = self._get_pp_attr("line", "length_km")
-        parallel = self._get_pp_attr("line", "parallel", 1)
-        c_nf_per_km = self._get_pp_attr("line", "c_nf_per_km")
-        c0_nf_per_km = self._get_pp_attr("line", "c0_nf_per_km", np.nan)
+        in_service = self._get_pp_attr("line", "in_service", expected_type="bool", default=True)
+        length_km = self._get_pp_attr("line", "length_km", expected_type="f8")
+        parallel = self._get_pp_attr("line", "parallel", expected_type="u4", default=1)
+        c_nf_per_km = self._get_pp_attr("line", "c_nf_per_km", expected_type="f8")
+        c0_nf_per_km = self._get_pp_attr("line", "c0_nf_per_km", expected_type="f8", default=np.nan)
         multiplier = length_km / parallel
 
         pgm_lines = initialize_array(data_type="input", component_type="line", shape=len(pp_lines))
         pgm_lines["id"] = self._generate_ids("line", pp_lines.index)
-        pgm_lines["from_node"] = self._get_pgm_ids("bus", self._get_pp_attr("line", "from_bus"))
+        pgm_lines["from_node"] = self._get_pgm_ids("bus", self._get_pp_attr("line", "from_bus", expected_type="u4"))
         pgm_lines["from_status"] = in_service & switch_states["from"]
-        pgm_lines["to_node"] = self._get_pgm_ids("bus", self._get_pp_attr("line", "to_bus"))
+        pgm_lines["to_node"] = self._get_pgm_ids("bus", self._get_pp_attr("line", "to_bus", expected_type="u4"))
         pgm_lines["to_status"] = in_service & switch_states["to"]
-        pgm_lines["r1"] = self._get_pp_attr("line", "r_ohm_per_km") * multiplier
-        pgm_lines["x1"] = self._get_pp_attr("line", "x_ohm_per_km") * multiplier
+        pgm_lines["r1"] = self._get_pp_attr("line", "r_ohm_per_km", expected_type="f8") * multiplier
+        pgm_lines["x1"] = self._get_pp_attr("line", "x_ohm_per_km", expected_type="f8") * multiplier
         pgm_lines["c1"] = c_nf_per_km * length_km * parallel * 1e-9
         # The formula for tan1 = R_1 / Xc_1 = (g * 1e-6) / (2 * pi * f * c * 1e-9) = g / (2 * pi * f * c * 1e-3)
         pgm_lines["tan1"] = (
-            self._get_pp_attr("line", "g_us_per_km", 0) / c_nf_per_km / (2 * np.pi * self.system_frequency * 1e-3)
+            self._get_pp_attr("line", "g_us_per_km", expected_type="f8", default=0)
+            / c_nf_per_km
+            / (2 * np.pi * self.system_frequency * 1e-3)
         )
-        pgm_lines["i_n"] = (self._get_pp_attr("line", "max_i_ka") * 1e3) * self._get_pp_attr("line", "df", 1) * parallel
-        pgm_lines["r0"] = self._get_pp_attr("line", "r0_ohm_per_km", np.nan) * multiplier
-        pgm_lines["x0"] = self._get_pp_attr("line", "x0_ohm_per_km", np.nan) * multiplier
+        pgm_lines["i_n"] = (
```

Comment:
@Jerry-Jinfeng-Guo From the error that is given for validation, maybe this assignment of i_n is putting a 0.0 instead of nan somewhere. Its failing because i_n should be either >0.0 or nan.

Reply:
For derating factor, default should be 1.0 as `exp_dtype = type(default) if default is not None else expected_type` is added. All defaults need to be checked to align with their type

Reply:
that still sounds error-prone. i would fix the issue in the implementation of the function, e.g.:

```py
# expected_type => default None instead of 'O'
# and then:
if expected_type is not None:
    exp_dtype = expected_type
elif default is not None:
    exp_dtype = type(default) if not isinstance(default, int) else float
else:
    exp_dtype = 'O'
```

Reply:
fixed in https://github.com/PowerGridModel/power-grid-model-io/pull/217/commits/2334bcaf4276ed7558d327742d1ea88c1ff09347